specification/agricultureplatform/AgriculturePlatform.DataPlane/models.tsp

```diff
@@ -0,0 +1,2140 @@
+import "@typespec/rest";
+import "@typespec/http";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace Microsoft.AgriculturePlatform;
+
+/**
+ * Type of relationship.
+ */
+union RelationshipType {
+  string,
+
+  /**
+   * DerivedFrom.
+   */
+  DerivedFrom: "DerivedFrom",
+
+  /**
+   * Contains.
+   */
+  Child: "Child",
+
+  /**
+   * Parent.
+   */
+  Parent: "Parent",
+}
+
+/**
+ * DestinationType.
+ */
+union DestinationType {
+  string,
+
+  /**
+   * Catalog.
+   */
+  Catalog: "Catalog",
+
+  /**
+   * Collection.
+   */
+  Collection: "Collection",
+
+  /**
+   * Item.
+   */
+  Item: "Item",
+
+  /**
+   * Asset.
+   */
+  Asset: "Asset",
+}
+
+/**
+ * Type of dataset.
+ */
+union DataFormat {
+  string,
+
+  /**
+   * Image.
+   */
+  Image: "Image",
+
+  /**
+   * Shapefile.
+   */
+  Shapefile: "Shapefile",
+
+  /**
+   * Binary
+   */
+  Binary: "Binary",
+}
+
+/**
+ * GeoJSON object type.
+ */
+union GeoJsonObjectType {
+  string,
+
+  /**
+   * Point.
+   */
+  Point: "Point",
+
+  /**
+   * Polygon.
+   */
+  Polygon: "Polygon",
+
+  /**
+   * MultiPolygon.
+   */
+  MultiPolygon: "MultiPolygon",
+}
+
+/**
+ * Enum for different types of AuthCredentials supported.
+ */
+union AuthCredentialsKind {
+  string,
+
+  /**
+   * OAuthClientCredentials.
+   */
+  OAuthClientCredentials: "OAuthClientCredentials",
+
+  /**
+   * ApiKeyAuthCredentials.
+   */
+  ApiKeyAuthCredentials: "ApiKeyAuthCredentials",
+}
+
+/**
+ * Supported image formats for scene resource.
+ */
+union ImageFormat {
+  string,
+
+  /**
+   * TIF.
+   */
+  TIF: "TIF",
+}
+
+/**
+ * CatalogListResponse.
+ */
+model CatalogListResponse {
+  /**
+   * List of Catalogs.
+   */
+  value: Catalog[];
+
+  /**
+   * Skip token for getting next set of results.
+   */
+  skipToken?: string;
+
+  /**
+   * URL to do the POST request with same filters,
+   * to get next set of items.
+   */
+  nextLink?: string;
+}
+
+/**
+ * Catalog.
+ */
+@resource("catalogs")
+model Catalog {
+  /**
+   * Links for referencing other objects.
+   */
+  links?: Link[];
+
+  /**
+   * Id of the Catalog.
+   */
+  @visibility("read")
+  @key("catalogId")
+  id: string;
+
+  /**
+   * The ETag value to implement optimistic concurrency.
+   */
+  @visibility("read")
+  eTag?: string;
+
+  /**
+   * Status of the resource.
+   */
+  @maxLength(100)
+  status?: string;
```

Comment:
Are there predefined values for status that can be defined in a `union` style enum?